src/Core/Timer/BasicTimer.cs

```diff
@@ -1,44 +1,143 @@
+        #region Explicitly implemented internal
+        void ITimer.Initialize(ITimeTrackingStrategy timeTrackingStrategy, List<ITimerTask> timerTasks)
+        {
+            Id = Guid.NewGuid();
+            foreach(ITimerTask timerTask in timerTasks)
+            {
+                // TODO: consider enforcing minimum period.
+                //
+                _timerTasks.Add(timerTask.Period, timerTask);
+            }
 
-        public void Clear()
+            _timeTrackingStrategy = timeTrackingStrategy;
+        }
+        public BasicTimer()
         {
-            throw new NotImplementedException();
+            _timerTasks = new SortedList<TimeSpan, ITimerTask>();
+            _currentlyRunning = new Dictionary<int, Task>();
+            _timeTrackingStrategy = null;
         }
 
-        public bool IsStopped()
+        /// <summary>
+        /// Schedules tasks based on ideal timings. For example, if a task has a period of
+        /// 1s but doesn't get run until T = 1.1s, the next schedule will be for T = 2s 
+        /// rather than 2.1s. Likewise for T = 0.9: second execution would be for T = 2s 
+        /// rather than 1.9.
+        /// 
+        /// Maintains a list of currently running tasks to potentially be cancelled later.
+        /// </summary>
+        /// <returns>Awaitable list of tasks running during this update.</returns>
+        async Task ITimer.Update()
         {
-            throw new NotImplementedException();
-        }
+            if(_timerTasks.Count == 0)
+            {
+                // TODO: Should we throw? Probably shouldn't have dangling Timers
+                //
+                await Task.CompletedTask;
+                return;
+            }
 
-        public void Pause()
+            _timeTrackingStrategy.Update();
+            TimeSpan elapsed = _timeTrackingStrategy.GetTimeElapsed();
+            TimeSpan nextTime = _timerTasks.Keys[0];
+
+            // Avoid creating on every update, as this might be happening very frequently.
+            //
+            List<KeyValuePair<TimeSpan, ITimerTask>> toAdd = null;
+            List<Task> runningTasks = null;
+
+            // NOTES: This works under the assumption we only want tasks to be able to be
+            // run once per update loop.
+            //
+            // Also, epsilon values allow for things to be run prior to their scheduled time.
+            // This value is intended to be very small and is meant to avoid tasks having
+            // to wait an entire update loop to run.
+            //
+            while (nextTime <= elapsed + _epsilon)
+            {
+                if(toAdd == null)
+                {
+                    toAdd = new List<KeyValuePair<TimeSpan, ITimerTask>>();
+                    runningTasks = new List<Task>();
+                }
+
+                ITimerTask taskToRun = _timerTasks.Values[0];
+                Action saveAndRun = async () =>
```

Comment:
have this as a private method which takes in an action and does the wrapping

something like Execute or ExecuteInternal